### PR TITLE
Add destination_subpath to setup-binary-test, fix Windows install_path

### PR DESCRIPTION
**Pull request by kevya-google**
Followup test for https://github.com/abcxyz/pkg/pull/270.

### CODE REVIEW

**Review by verbanicm — DISMISSED** [review]

File: .github/workflows/setup-binary-test.yml [hunk 1]
```diff
@@ -67,11 +67,12 @@ jobs:
...
           binary_subpath: 'ratchet'
+          destination_subpath: 'renamed_ratchet'
 
       - name: 'Test'
         shell: 'bash'
         run: |
-          ratchet -version
+          renamed_ratchet -version
           GOT=$(ratchet -version 2>&1) # abcxyz/pkg/cli writes help to stderr
           WANT="ratchet 0.6.0"
```
Comment (verbanicm) [inline .github/workflows/setup-binary-test.yml:77]:
```suggestion
          renamed_ratchet -version
          GOT=$(renamed_ratchet -version 2>&1)
          WANT="ratchet 0.6.0"
```

File: .github/actions/setup-binary/action.yml [hunk 2]
```diff
@@ -142,6 +142,7 @@ runs:
           DOWNLOAD_URL="${DOWNLOAD_URL}${SEPARATOR}checksum=${CHECKSUM}"
         fi
 
+        mkdir -p ${INSTALL_PATH}
```
Comment (verbanicm) [inline .github/actions/setup-binary/action.yml:145]: I think you'll want mkdir -p here

File: .github/workflows/setup-binary-test.yml [hunk 3]
```diff
@@ -43,19 +43,20 @@ jobs:
       - name: 'Checkout'
         uses: 'actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11' # ratchet:actions/checkout@v4
 
-      - name: 'Path'
+      - name: 'OS friendly path'
         id: 'path'
-        shell: 'bash'
+        shell: 'pwsh'
```
Comment (lock14) [inline .github/workflows/setup-binary-test.yml:48]: why are we switching the shell to PowerShell from bash?

Comment (kevya-google) [reply .github/workflows/setup-binary-test.yml:48]: nvm, switching back to bash.

Comment (kevya-google) [reply .github/workflows/setup-binary-test.yml]: nvm, switched back to powershell but just for windows lol

**Review by verbanicm — APPROVED** [review]

**Pull request merged**